contract/src/test/java/io/neow3j/contract/TransactionBuilderTest.java

```diff
@@ -84,137 +86,254 @@ public void setUp() {
+
+        Transaction t = b.nonce(nonce).buildTransaction();
+        assertThat(t.getNonce(), is(nonce));
+
+        nonce = 0L;
+        t = b.nonce(0L).buildTransaction();
+        assertThat(t.getNonce(), is(nonce));
+
+        nonce = (long) Math.pow(2, 32) - 1;
+        t = b.nonce(nonce).buildTransaction();
+        assertThat(t.getNonce(), is(nonce));
+
+        nonce = Integer.toUnsignedLong(-1);
+        t = b.nonce(nonce).buildTransaction();
+        assertThat(t.getNonce(), is(nonce));
     }
 
     @Test
-    public void failWithoutSettingWallet() throws IOException {
-        Invocation.Builder b = new Invocation.Builder(neow)
-                .withContract(NEO_TOKEN_SCRIPT_HASH)
-                .withFunction("name");
-        exceptionRule.expect(InvocationConfigurationException.class);
-        exceptionRule.expectMessage("wallet");
-        b.build();
+    public void failBuildingTransactionWithIncorrectNonce() {
+        TransactionBuilder b = new TransactionBuilder(neow)
+                .validUntilBlock(1L)
+                .wallet(Wallet.withAccounts(account1))
+                .script(new byte[]{1, 2, 3})
+                .signers(Signer.calledByEntry(account1.getScriptHash()));
+        try {
+            Long nonce = Integer.toUnsignedLong(-1) + 1;
+            b.nonce(nonce);
+            fail();
+        } catch (TransactionConfigurationException ignored) {
+        }
+
+        try {
+            Long nonce = (long) Math.pow(2, 32);
+            b.nonce(nonce);
+            fail();
+        } catch (TransactionConfigurationException ignored) {
+        }
+
+        try {
+            Long nonce = -1L;
+            b.nonce(nonce);
+            fail();
+        } catch (TransactionConfigurationException ignored) {
+        }
+    }
+
+    @Test(expected = TransactionConfigurationException.class)
+    public void failBuildingTransactionWithNegativeValidUntilBlockNumber() {
+        new TransactionBuilder(neow)
+                .validUntilBlock(-1L)
+                .wallet(Wallet.withAccounts(account1))
+                .script(new byte[]{1, 2, 3})
+                .signers(Signer.calledByEntry(account1.getScriptHash()));
+    }
+
+    @Test(expected = TransactionConfigurationException.class)
+    public void failBuildingTransactionWithTooHighValidUntilBlockNumber() {
+        new TransactionBuilder(neow)
+                .validUntilBlock((long) Math.pow(2, 32))
+                .wallet(Wallet.withAccounts(account1))
+                .script(new byte[]{1, 2, 3})
+                .signers(Signer.calledByEntry(account1.getScriptHash()));
     }
 
     @Test
-    public void testAutomaticSettingOfValidUntilBlockVariable() throws IOException {
-        Wallet wallet = Wallet.createWallet();
-        setUpWireMockForCall("invokescript", "invokescript_name_neo.json", SCRIPT);
-        ContractTestHelper.setUpWireMockForGetBlockCount(1000);
+    public void automaticallyFetchValidUntilBlock() throws Throwable {
```

Comment:
This doesn't test what it says in the method name

Reply:
You're right, I renamed it to `automaticallySetNonce`.